### PR TITLE
added mirroring blog

### PR DESCRIPTION
**Pull request by christian-posta**
cc @rshriram @thisisnotapril

### CODE REVIEW

**Comment by rshriram** [comment]
I am going to let @frankbu / @geeknoid take a crack at this. We are trying to keep the language/tone consistent across our blogs.

File: _blog/2018/traffic-mirroring.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+title: "Traffic mirroring with Istio for testing in production"
+overview: An introduction to safer, lower-risk deployments and release to production
+publish_date: February 8, 2018
+subtitle: Routing rules for HTTP traffic
+attribution: Christian Posta
+
+order: 95
```
Comment (geeknoid) [inline _blog/2018/traffic-mirroring.md]: Should be 91 (one less than the current latest post)

File: _blog/2018/traffic-mirroring.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+layout: blog
+type: markdown
+redirect_from: "/blog/traffic-mirroring.html"
+---
+{% include home.html %}
+
+Tryign to enumerate all of the possible cominations of test cases for testing services in lower environments can be daunting. In some cases, you'll find that all of the effort that goes into cataloging this use cases doesn't match up to real production use cases. Ideally, we could use live production usecases and traffic to help illuminate all of the areas of our code that we might miss in more contrived lower-stage testing.
```
Comment (geeknoid) [inline _blog/2018/traffic-mirroring.md]: Tryign -> Trying
cominations -> combinations
usecases -> use cases

Comment (geeknoid) [reply _blog/2018/traffic-mirroring.md]: What's "a lower environment"?
What's "lower-stage"?

File: _blog/2018/traffic-mirroring.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+layout: blog
+type: markdown
+redirect_from: "/blog/traffic-mirroring.html"
+---
+{% include home.html %}
+
+Tryign to enumerate all of the possible cominations of test cases for testing services in lower environments can be daunting. In some cases, you'll find that all of the effort that goes into cataloging this use cases doesn't match up to real production use cases. Ideally, we could use live production usecases and traffic to help illuminate all of the areas of our code that we might miss in more contrived lower-stage testing. 
+
+Istio can help here. With the release of [Istio 0.5.0]({{home}}/about/notes/0.5.html), Istio can mirror traffic to help test your services. We can write route rules similar to the following the enable mirroring:
```
Comment (geeknoid) [inline _blog/2018/traffic-mirroring.md]: We can -> You can

File: _blog/2018/traffic-mirroring.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    name: httpbin
+    labels:
+      version: v2
+``` 
+
+A few things to note here:
+
+* When traffic gets mirrored to a different service, that happens outside the critical path of the request
+* Responses to any mirrored traffic is ignored; traffic is mirrored as "fire-and-forget"
+* We'll need to have the 0 weighted route to hint to Istio to create the proper Envoy cluster under the covers; [this should be ironed out in future releases](https://github.com/istio/istio/issues/3270).
```
Comment (geeknoid) [inline _blog/2018/traffic-mirroring.md]: We'll need -> You need

File: _blog/2018/traffic-mirroring.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+For more on this, see the [Mirroring Task]({{home}}/docs/tasks/traffic-management/mirroring.html) from the Istio docs and see the more [comprehensive treatment of this scenario on my blog](http://blog.christianposta.com/microservices/traffic-shadowing-with-istio-reduce-the-risk-of-code-release/)
```
Comment (geeknoid) [inline _blog/2018/traffic-mirroring.md]: Learn more about mirroring by visiting [Mirroring Task]({{home}}/docs/tasks/traffic-management/mirroring.html),and see the more [comprehensive treatment of this scenario on my blog](http://blog.christianposta.com/microservices/traffic-shadowing-with-istio-reduce-the-risk-of-code-release/)

**Comment by christian-posta** [comment]
thx @geeknoid updated with latest commit

**Comment by thisisnotapril** [comment]
Thanks for submitting this @christian-posta!

File: _blog/2018/traffic-mirroring.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      version: v2
+``` 
+
+A few things to note here:
+
+* When traffic gets mirrored to a different service, that happens outside the critical path of the request
+* Responses to any mirrored traffic is ignored; traffic is mirrored as "fire-and-forget"
+* You'll need to have the 0-weighted route to hint to Istio to create the proper Envoy cluster under the covers; [this should be ironed out in future releases](https://github.com/istio/istio/issues/3270).
+
+Learn more about mirroring by visiting the [Mirroring Task]({{home}}/docs/tasks/traffic-management/mirroring.html) from the Istio docs and see the more [comprehensive treatment of this scenario on my blog](http://blog.christianposta.com/microservices/traffic-shadowing-with-istio-reduce-the-risk-of-code-release/).
```
Comment (geeknoid) [inline _blog/2018/traffic-mirroring.md]: No need to say "from the Istio docs". This is istio.io already...

File: _blog/2018/traffic-mirroring.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+order: 91
+
+layout: blog
+type: markdown
+redirect_from: "/blog/traffic-mirroring.html"
+---
+{% include home.html %}
+
+Trying to enumerate all of the possible combinations of test cases for testing services in non-production/test environments can be daunting. In some cases, you'll find that all of the effort that goes into cataloging these use cases doesn't match up to real production use cases. Ideally, we could use live production use cases and traffic to help illuminate all of the areas of our code that we might miss in more contrived testing environments.
```
Comment (geeknoid) [inline _blog/2018/traffic-mirroring.md]: all of the -> all the

Comment (geeknoid) [reply _blog/2018/traffic-mirroring.md]: Can we find a better way to say "all of the areas of our code". FOr example, "illuminate all of the feature areas of the service under test"

**Comment by ldemailly** [comment]
it seems to be quite similar / little additional content compared to https://istio.io/docs/tasks/traffic-management/mirroring.html (and maybe updates should be made to that task directly instead?)

on the other hand I suppose blogs highlighting new features can be nice to have

**Pull request merged**